.github/workflows/tekton_bundle_push.yaml

```diff
@@ -0,0 +1,127 @@
+---
+name: Tekton Bundle Push
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches: ['main']
+  workflow_dispatch:
+env:
+  IMAGE_REGISTRY: quay.io
+  IMAGE_NAMESPACE: hacbs-release
+  REGISTRY_USER: ${{ secrets.QUAY_ROBOT_USER }}
+  REGISTRY_PASSWORD: ${{ secrets.QUAY_ROBOT_TOKEN }}
+  API_TOKEN: ${{ secrets.QUAY_API_TOKEN }}
+jobs:
+  run-pipeline:
+    name: Tekton Bundle
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2  # using git diff-tree requires fetch depth=2
+      - run: |
+          mkdir -vp $HOME/.kube || true
+          cat <<-EOF > $HOME/.kube/config
+          apiVersion: v1
+          kind: Config
+          clusters:
+          - cluster:
+              server: _
+            name: _
+          contexts:
+          - context:
+              cluster: _
+            name: _
+          current-context: _
+          EOF
+      - uses: jerop/tkn@v0.1.0
+      - uses: redhat-actions/podman-login@v1
+        with:
+          username: ${{ env.REGISTRY_USER }}
+          password: ${{ env.REGISTRY_PASSWORD }}
+          registry: ${{ env.IMAGE_REGISTRY }}
+      - run: |
+          printf 'Gathering new/modified bundle definitions.\n'
+
+          # Gather new/modified bundle definitions.
+          declare -A bundle_dirs  # associative array to hash bundle dirs in key space.
+          while read file
+          do
+            # Only match yaml files in bundle dirs within definitions dir.
+            if [[ $file =~ .*[-,_]{2,} ]]
+            then
+              printf 'NO consecutive hyphen or underscores allowed: %s\n' "$file"
+            elif [[ ${file} =~ ^("./")?"definitions/"[a-z,A-Z,0-9,_-]+"/"[a-z,A-Z,0-9,_-]+".yaml"$ ]]
+            then
+              printf 'MATCHED: %s\n' "$file"
+              bundle_dirs["${file%/*}"]=""  # hash the bundle dir
+            else
+              printf 'NO MATCH: %s\n' "$file"
+            fi
+          done < <(git diff-tree --no-commit-id --name-only -r ${{ github.sha }})
+          # EO gathering changed definitions
+          printf 'bundle_dirs: %s\n' "${!bundle_dirs[@]}"
+
+          # Main loop
+          printf 'Pushing any new/modified bundle definitions.\n'
+          for BUNDLE_DIR in "${!bundle_dirs[@]}"
+          do
+            if [[ ! -d "$BUNDLE_DIR" ]]
+            then
+              printf 'Skipping non-existing bundle-path: %s\n' "$BUNDLE_DIR"
+              continue
+            fi
+            BUNDLE_FILES=()
+            BUNDLE_NAME=${BUNDLE_DIR##*/}
+            printf '* Bundle_dir: %s\n' $BUNDLE_DIR
+            printf '* Bundle name: %s\n' $BUNDLE_NAME
+            for file in $BUNDLE_DIR/*.yaml
+            do
+              [[ -e "$file" ]] || continue
+              printf '  * file: %s\n' "$file"
+              BUNDLE_FILES+=("$file")
+            done
+
+            # xxx - tkn cli is apparently unable to process a list of bundle files on one `-f` cmdline arg.
```

Comment:
Not strictly true. You can pass a file with all the resources separated by `---` and all of them will get processed. It works exactly in the same way than kubectl/oc.

Reply:
To clarify, 
What does NOT work is this hypothetical example, where you may assume there are multiple YAML files:
```
tkn bundle push  -f ./definition/hello-world/*.yaml  "hello-world:main"
```